modules/http4s-server/src/main/scala/io/janstenpickle/trace4cats/http4s/server/ServerSyntax.scala

```diff
@@ -14,9 +17,19 @@ trait ServerSyntax {
       spanNamer: Http4sSpanNamer = Http4sSpanNamer.methodWithPath,
       requestFilter: Http4sRequestFilter = Http4sRequestFilter.allowAll,
       dropHeadersWhen: CaseInsensitiveString => Boolean = Headers.SensitiveHeaders.contains
-    )(implicit F: Bracket[F, Throwable], provide: Provide[F, G], lift: LiftTrace[F, G]): HttpRoutes[F] =
+    )(implicit P: Provide[F, G, Span[F]], F: Bracket[F, Throwable]): HttpRoutes[F] =
       ServerTracer
-        .injectRoutes[F, G](routes, entryPoint, spanNamer, requestFilter, dropHeadersWhen)
+        .injectRoutes(routes, entryPoint, spanNamer, requestFilter, dropHeadersWhen, _ => Kleisli.ask[F, Span[F]])
```

Comment:
Both here and below you've introduced a dependency on Kleisli for this syntax, was this deliberate?

Reply:
Agree, it was too much.

We need a builder function for `Ctx` from `Request_` and `Span[F]`. 
Let its signature be `(Request_, Span[F]) => F[Ctx]`

Users that have a compound context will need to create such a function and provide it to the syntax extension, e.g.
```scala
def mkAppCtx[F[_] : MonadThrow](req: Request_, span: Span[F]): F[AppCtx[F]] = ???

ctxRoutes.injectContext(entryPoint, makeContext = mkAppCtx[F])
```

Reply:
Or we can choose another signature, e.g. `Request_ => Span[F] => F[Ctx]`

It will affect the way a user defines their maker function.

Reply:
We can even define an alias for this.

What do you think?

Reply:
Also, it may have a better user experience in a curried form:
```scala
def injectContext[Ctx](
  entryPoint: EntryPoint[F],
  spanNamer: Http4sSpanNamer = Http4sSpanNamer.methodWithPath,
  requestFilter: Http4sRequestFilter = Http4sRequestFilter.allowAll,
  dropHeadersWhen: CaseInsensitiveString => Boolean = Headers.SensitiveHeaders.contains
)(makeContext: (Request_, Span[F]) => F[Ctx])
```

Reply:
I just had a chance to check this out look this over properly in my IDE and what you've done without the currying looks good.